lottie/src/test/java/com/airbnb/lottie/LottieCompositionFactoryTest.java

```diff
@@ -83,11 +86,20 @@ public void testNonJsonAssetFile() {
 
     @Test
     public void testLoadInvalidRawResName() {
-        LottieResult<LottieComposition> result = LottieCompositionFactory.fromRawResSync(RuntimeEnvironment.application, 0);
+        WeakReference<Context> contextRef = new WeakReference<Context>(RuntimeEnvironment.application);
+        LottieResult<LottieComposition> result = LottieCompositionFactory.fromRawResSync(contextRef, 0);
         assertNotNull(result.getException());
         assertNull(result.getValue());
     }
 
+    @Test
+    public void testContextNotReachable() {
+        WeakReference<Context> contextRef = new WeakReference<>(null);
+        LottieResult<LottieComposition> result = LottieCompositionFactory.fromRawResSync(contextRef, 0);
+        assertTrue(result.getException() instanceof NullPointerException);
```

Comment:
This behavior could introduce a regression since it will fail to parse in some cases where it would succeed today.
Can you make this hold the weak Activity and Application context and if Activity context is gone, default to day mode but continue to parse and cache it?

Reply:
Sure! Forcepushed the change.